modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/MessageBoardThreadResourceImpl.java

```diff
@@ -794,7 +794,8 @@ private void _updateQuestion(
 		}
 
 		if (GetterUtil.getBoolean(messageBoardThread.getSubscribed())) {
-			_mbMessageService.subscribeMessage(mbMessage.getRootMessageId());
+			_mbMessageLocalService.subscribeMessage(
```

Comment:
There are two methods calling this _updateQuestion private method.

- First case

The first one is the add method, which creates a message and then subscribes to it. 

We can infer that if we can create the message and we are the owners we should be able to subscribe to it. Perfect in this case

- Second case

The second one is the update method, here it gets the message, calls this method to subscribe to it, and then updates it. 

Here we are considering that if we can see the message we can subscribe to it but I'm not so comfortable with this supposition, because, in reality, they are different permissions. 

I will be more comfortable if we follow the same pattern as add, first updating the message and saving it in a local variable, then subscribing to it and finally returning the DTO. This way we can infer that if we can update the message we can subscribe to it that it is a stronger supposition. 

Also, we create a consistent pattern between add and update

Reply:
Yes! In fact, in my description, I supposed that first, we update the message, and then we subscribe to it. My bad because didn't realize that the update method does not follow that pattern. Thanks, Javi 🙂 
![](https://media.giphy.com/media/69lPHFKRRW9xdwzDfa/giphy.gif)